src/main/java/capone/Parser.java

```diff
@@ -0,0 +1,169 @@
+        } else if (firstWord.equalsIgnoreCase("bye")) {
+            return new ByeCommand(scanner);
+        } else if (firstWord.equalsIgnoreCase("help")) {
+            return new HelpCommand();
+        } else {
+            return new InvalidCommand();
+        }
+    }
+
+    /**
+     * Checks if the given input string follows the date format 'YYYY-MM-DD'.
+     *
+     * @param input The input string to be checked.
+     * @return True if the input matches the date format, false otherwise.
+     */
+    public static boolean isDateFormat(String input) {
+        String dateFormatRegex = "\\d{4}-\\d{2}-\\d{2}";
+        return input.matches(dateFormatRegex);
+    }
+
+    /**
+     * Checks if the given input string follows the time
+     * format 'HHmm' (24-hour format).
+     *
+     * @param input The input string to be checked.
+     * @return True if the input matches the time format, false otherwise.
+     */
+    public static boolean isTimeFormat(String input) {
+        String timeFormatRegex = "(\\d{4})";
+        return input.matches(timeFormatRegex);
+    }
+
+    /**
+     * Processes the date and time components and returns a LocalDateTime object.
+     *
+     * @param date The LocalDate object representing the date.
+     * @param time The LocalTime object representing the time.
+     * @return LocalDateTime object combining the date and time.
+     */
+    public static LocalDateTime processDateTime(LocalDate date, LocalTime time) {
+        if (date != null) {
+            if (time != null) {
+                return date.atTime(time);
+            } else {
+                return date.atStartOfDay();
+            }
+        } else {
+            if (time != null) {
+                return LocalDate.now().plusDays(1).atTime(time);
+            } else {
+                return null;
+            }
+        }
+    }
+
+    /**
+     * Parses the input string into a LocalDate object.
+     *
+     * @param date The input string representing the date.
+     * @return LocalDate object parsed from the input.
+     * @throws InvalidDateException If the input string is not a valid date.
+     */
+    public static LocalDate parseDate(String date) throws InvalidDateException {
+        DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+        try {
+            return LocalDate.parse(date, dateFormatter);
+        } catch (DateTimeException e) {
+            throw new InvalidDateException("Oops! You have entered an invalid date. Please try again.");
+        }
+    }
+
+    /**
+     * Parses the input string into a LocalTime object.
+     *
+     * @param time The input string representing the time.
+     * @return LocalTime object parsed from the input.
+     * @throws InvalidTimeException If the input string is not a valid time.
+     */
+    public static LocalTime parseTime(String time) throws InvalidTimeException {
+        DateTimeFormatter timeFormatter = DateTimeFormatter.ofPattern("HHmm");
```

Comment:
Perhaps you could also include the input where the user types in HH:mm as that is also a common way of writing the time

Reply:
Yes, that's a good idea. I will consider your suggestion!